### ISSUE TITLE
Product is not removed from the website during the import

### ISSUE THREAD

**Issue by lyusya-nnn** [magento/magento2#22427]
### Preconditions (*)

 Magento 2.3.1 with 4 websites (website codes base, b2b, b2r and b2c)

### Steps to reproduce (*)

1. Create a simple product with sku **test1** that is attached to all websites
2. Create a csv file with next content:
sku,product_websites
test1,b2b
3. Go to System -> Import, select Entity Type = Products, Import Behavior = Add/Update, Select File to Import = upload your CSV file and run import
4. Recheck your product test1

### Expected result (*)

Product test1 is attached only to b2b website

### Actual result (*)

product test1 is still attached to all websites

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99321` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by gulshanchitransh** [comment]
I am working on this at #dmcdindia1

**Comment by PascalBrouwers** [comment]
@lyusya-nnn I case you are wondering this is because of `vendor/magento/module-catalog-import-export/Model/Import/Product.php` function `_saveProductWebsites`'s code:
```
if (Import::BEHAVIOR_APPEND != $this->getBehavior()) {
                $this->_connection->delete(
                    $tableName,
                    $this->_connection->quoteInto('product_id IN (?)', $delProductId)
                );
            }
```

Apparently no websites are removed during append action. I think this is totally illogical too, but you need to use a different behaviour like `BEHAVIOR_ADD_UPDATE` or `BEHAVIOR_REPLACE`.

**Comment by magento-engcom-team** [comment]
Hi @lyusya-nnn. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-24376` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
